core/auth/fake/interfaces/controller_test.go

```diff
@@ -0,0 +1,397 @@
+			fields: fields{
+				config: fakeConfig{
+					UserConfig: map[string]userConfig{
+						"user_a": userConfig{
+							Password: "test_a",
+							Otp:      "otp_a",
+						},
+						"user_b": userConfig{
+							Password: "test_b",
+							Otp:      "otp_b",
+						},
+					},
+					ValidatePassword: true,
+				},
+			},
+			args: args{
+				r: addRequestParameters(
+					web.CreateRequest(
+						&http.Request{
+							Method: http.MethodPost,
+							URL: &url.URL{
+								Scheme: "http",
+							},
+							PostForm: url.Values{
+								"username": []string{"user_b"},
+								"password": []string{"test_b"},
+							},
+						},
+						web.EmptySession(),
+					),
+				),
+			},
+			want: &web.RouteRedirectResponse{
+				Response: web.Response{
+					Status: http.StatusSeeOther,
+					Header: http.Header{},
+				},
+				To:   "core.auth.callback",
+				Data: map[string]string{"broker": "testBroker"},
+			},
+		},
+		{
+			name: "login for valid user with password, otp mismatch",
+			fields: fields{
+				config: fakeConfig{
+					UserConfig: map[string]userConfig{
+						"user_a": userConfig{
+							Password: "test_a",
+							Otp:      "otp_a",
+						},
+						"user_b": userConfig{
+							Password: "test_b",
+							Otp:      "otp_b",
+						},
+					},
+					ValidatePassword: true,
+					ValidateOtp:      true,
+				},
+			},
+			args: args{
+				r: addRequestParameters(
+					web.CreateRequest(
+						&http.Request{
+							Method: http.MethodPost,
+							URL: &url.URL{
+								Scheme: "http",
+							},
+							PostForm: url.Values{
+								"username": []string{"user_b"},
+								"password": []string{"test_b"},
+								"otp":      []string{"invalid otp"},
+							},
+						},
+						web.EmptySession(),
+					),
+				),
+			},
+			want: wantFormResponseWithMessage(errOtpMismatch),
+		},
+		{
+			name: "login for valid user / valid password / valid otp",
+			fields: fields{
+				config: fakeConfig{
+					UserConfig: map[string]userConfig{
+						"user_a": userConfig{
+							Password: "test_a",
+							Otp:      "otp_a",
+						},
+						"user_b": userConfig{
+							Password: "test_b",
+							Otp:      "otp_b",
+						},
+					},
+					ValidatePassword: true,
+					ValidateOtp:      true,
+				},
+			},
+			args: args{
+				r: addRequestParameters(
+					web.CreateRequest(
+						&http.Request{
+							Method: http.MethodPost,
+							URL: &url.URL{
+								Scheme: "http",
+							},
+							PostForm: url.Values{
+								"username": []string{"user_b"},
+								"password": []string{"test_b"},
+								"otp":      []string{"otp_b"},
+							},
+						},
+						web.EmptySession(),
+					),
+				),
+			},
+			want: &web.RouteRedirectResponse{
+				Response: web.Response{
+					Status: http.StatusSeeOther,
+					Header: http.Header{},
+				},
+				To:   "core.auth.callback",
+				Data: map[string]string{"broker": "testBroker"},
+			},
+		},
+	}
+
+	for _, tt := range tests {
```

Comment:
Do the table tests bring us any value here, or do they limit is in ways that we need to overspecify testdata and are limited in what we can assert?